core/src/worker/mod.rs

```diff
@@ -471,7 +509,7 @@ impl Worker {
         // about to happen anyway. Tell the local activity manager that, so that it can know to
         // cancel any remaining outstanding LAs and shutdown.
         if matches!(r, Err(PollWfError::ShutDown)) {
-            self.initiate_shutdown();
+            self.initiate_shutdown(); // TODO: Why is this here? Wasn't this supposed to have been called already?
```

Comment:
This is covering the situation where WFT pollers dying is the reason for shutdown